component/src/main/java/org/cbioportal/genome_nexus/component/annotation/IndexResolver.java

```diff
@@ -0,0 +1,158 @@
+            for (TranscriptConsequence transcriptConsequence : variantAnnotation.getTranscriptConsequences()) {
+                if (transcriptConsequence.getGeneSymbol() != null) {
+                    hugoSymbolSet.add(transcriptConsequence.getGeneSymbol());
+                }
+            }
+            hugoSymbol = hugoSymbolSet.stream().collect(Collectors.toList());
+        }
+
+        return hugoSymbol;
+    }
+
+    @Nullable
+    private List<String> resolveHgvsp(VariantAnnotation variantAnnotation)
+    {
+        List<String> hgvsp = null;
+
+        if (variantAnnotation != null &&
+            variantAnnotation.getTranscriptConsequences() != null &&
+            variantAnnotation.getTranscriptConsequences().size() > 0)
+        {
+            Set<String> hgvspSet = new HashSet<>();
+            for (TranscriptConsequence transcriptConsequence : variantAnnotation.getTranscriptConsequences()) {
+                if (transcriptConsequence.getHgvsp() != null) {
+                    hgvspSet.add(transcriptConsequence.getHgvsp());
+                }
+            }
+            hgvsp = hgvspSet.stream().collect(Collectors.toList());
+        }
+
+        return hgvsp;
+    }
+
+    @Nullable
+    private List<String> resolveHgvsc(VariantAnnotation variantAnnotation)
+    {
+        List<String> hgvsc = null;
+
+        if (variantAnnotation != null &&
+            variantAnnotation.getTranscriptConsequences() != null &&
+            variantAnnotation.getTranscriptConsequences().size() > 0)
+        {
+            Set<String> hgvscSet = new HashSet<>();
+            for (TranscriptConsequence transcriptConsequence : variantAnnotation.getTranscriptConsequences()) {
+                if (transcriptConsequence.getHgvsc() != null) {
+                    hgvscSet.add(transcriptConsequence.getHgvsc());
+                }
+            }
+            hgvsc = hgvscSet.stream().collect(Collectors.toList());
+        }
+
+        return hgvsc;
+    }
+
+    @Nullable
+    private List<String> resolveHgvspShort(VariantAnnotation variantAnnotation)
+    {
+        List<String> hgvspShort = null;
+
+        if (variantAnnotation != null &&
+            variantAnnotation.getTranscriptConsequences() != null &&
+            variantAnnotation.getTranscriptConsequences().size() > 0)
+        {
+            Set<String> hgvspShortSet = new HashSet<>();
+            for (TranscriptConsequence transcriptConsequence : variantAnnotation.getTranscriptConsequences()) {
+                String hgvspShortResult = this.proteinChangeResolver.resolveHgvspShort(variantAnnotation, transcriptConsequence);
+                if (hgvspShortResult != null) {
+                    hgvspShortSet.add(hgvspShortResult);
+                }
+            }
+            hgvspShort = hgvspShortSet.stream().collect(Collectors.toList());
+        }
+
+        return hgvspShort;
+    }
+
+    @Nullable
+    private List<String> resolveCdna(VariantAnnotation variantAnnotation)
```

Comment:
Not sure where exactly to move this method, but similar to the other ones, it might be better to have this one in another resolver too.